### ISSUE TITLE
unpaper dies via SIGKILL: 9 - Stream #0: not enough frames to estimate rate

### ISSUE THREAD

**Issue by tedtms** [unpaper/unpaper#96]
**Describe the bug**:
unpaper dies via SIGKILL: 9 with the following errors when processing a file:

```
[ppm_pipe @ 0x5583d136b9c0] Stream #0: not enough frames to estimate rate; consider increasing probesize
[image2 @ 0x5583d1371040] Using AVStream.codec to pass codec parameters to muxers is deprecated, use AVStream.codecpar instead.
```

**To Reproduce**:
`unpaper -v --dpi 599.6826446280992 --layout none --mask-scan-size 100 --no-border-align --no-mask-center --no-grayfilter --no-blackfilter --no-deskew /tmp/tmp1b9ca3lu/input.pnm /tmp/tmp1b9ca3lu/output.ppm`

**Logs**:
I'm using unpaper when it is called by ocrmypdf. Please find the attached log: 
[ocrmypdf.log](https://github.com/unpaper/unpaper/files/6527249/ocrmypdf.log)

**System**:
    OS: Ubuntu 20.04
    Python version: 3.8.5
    OCRmyPDF version: 12.0.1
    unpaper version: 6.1

**Installation method**:
apt

**Additional context**:
I encountered this error while using [paperless-ng](https://github.com/jonaswinkler/paperless-ng/issues/1045) and was directed to use ocrmypdf standalone to try to reproduce the issue. To do so, I've attempted to run the same arguments that paperless-ng invokes while processing a document:

```ocrmypdf -v -d -r -c --output-type pdfa -j 2 -l eng --sidecar /tmp/paperless-xzsn3/sidecar.txt brother_05222021_152203_001128.pdf /tmp/paperless/paperless-xzsn3_6x/archive.pdf 2>&1 | tee ~/ocrmypdf.log```

As shown in the attached log, ocrmypdf correctly saves the temporary files and passes them to unpaper. Unpaper then experiences the pipeline error. FWIW, I opened the issue originally with paperless-ng, who directed me to ocrmypdf, who then directed me here. Any help you can offer would be greatly appreciated!

**Comment by Flameeyes** [comment]
Hi, is there any hope you can share the file that was being processed? If not it makes difficult for me to reproduce the problem — this appears to be an FFmpeg issue in the sense that the file is not being parsed correctly, so without an example file I can't easily do much about this.

**Comment by tedtms** [comment]
So I can't give you the original problem document, but I'm experiencing this issue with any document that requires OCR so I tried it on this one and got the same result.
[brother_05232021_191033_001129.pdf](https://github.com/unpaper/unpaper/files/6529148/brother_05232021_191033_001129.pdf)

**Comment by Flameeyes** [comment]
Thanks — I'll need to figure out how to get from this to whatever is being passed to unpaper, so will let you know when I can reproduce. (unpaper doesn't read PDF natively.)

**Comment by Flameeyes** [comment]
Also, can you try with an unpaper built at the current main head? — I can't seem to reproduce with that, so I might just as well get off my laziness and make a new release if that works fine.

**Comment by tedtms** [comment]
Hey, just bumping this to check status -- were you able to reproduce my issue? Is there anything else I can provide to help?

I ended up rebuilding my entire paperless-ng environment in case I missed some sort of dependency or other requirement, but I ran into the same exact error with unpaper in the same exact place in the paperless workflow.

**Comment by tedtms** [comment]
So, I actually figured it out. I've been running paperless-ng (and therefore, unpaper) in an Ubuntu 20.04 _proxmox LXC container_. For all intents and purposes, this behaves the same way as running a full-blown VM. Up until the most recent release(s), it had been running fine on 512MB of virtual memory. But by pure happenstance, I had my proxmox server's console monitor pulled up this morning and noticed an out of memory error on the main terminal:
```
Memory cgroup out of memory: Killed process 14395 (unpaper) total-vm:637688kB, anon-rss:371712kB, file-rss:0kB shmem-rss:0kB, UID:101104 pgtables:1160kB oom_score_adj:0
```
UID 101104 is how proxmox views my container's paperless user, and obviously unpaper is running on that container as well. For those unaware, LXC containers (and really, all others) use the main OS's kernel and abstract it for the containers. So these _kernel level_ out of memory errors were occurring on the main platform and not the actual container, so I never got an out of memory container _inside_ my paperless-ng instance.

After seeing this error, I upped my container's virtual memory to 1GB, and re-consumed a document. Lo and behold, it consumed without issue. So! I would highly recommend that anyone experiencing this error look at upping either their machine's or their container's memory. :)
